Source/Dafny/Compiler-java.cs

```diff
@@ -1691,13 +1691,14 @@ protected class ClassWriter : IClassWriter {
       return true;
     }
 
-    protected string GetClassPath(string targetFilename) {
+    private static string GetClassPath(string targetFilename) {
       var assemblyLocation = Assembly.GetExecutingAssembly().Location;
       Contract.Assert(assemblyLocation != null);
       var codebase = Path.GetDirectoryName(assemblyLocation);
       Contract.Assert(codebase != null);
-      // DafnyRuntime-1.jar has already been created using Maven. It is added to the java CLASSPATH below.
-      return "." + Path.PathSeparator + Path.GetFullPath(Path.GetDirectoryName(targetFilename)) + Path.PathSeparator + Path.Combine(codebase, "DafnyRuntime-1.jar");
+      var targetPath = DafnyOptions.O.JavaCompileTargetDirectory ?? Path.GetFullPath(Path.GetDirectoryName(targetFilename));
+      var paths = new List<string> { ".", targetPath, Path.Combine(codebase, "DafnyRuntime-1.jar") };
```

Comment:
Can we rename the file to just `DafnyRuntime.jar`?

Reply:
The "1" refers to the version of DafnyRuntime, which was chosen to be 1 by the interns who did the Java compiler, and apparently [still is 1](https://github.com/dafny-lang/dafny/blob/de4eed9308aa1d3af45426ed2970e3c429bfe61a/Source/DafnyRuntime/DafnyRuntimeJava/build.gradle#L18) in the new gradle-based build. 
Java build systems like JARs named `PackageName-VersionNumber.jar`, but I'm not sure if this copy of the JAR will be seen by build systems, so I don't have solutions here, just something I wanted to point out in case it helps.